flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -1625,6 +1626,17 @@ IntrinsicClass IntrinsicProcTable::Implementation::GetIntrinsicClass(
   return IntrinsicClass::noClass;
 }
 
+std::string IntrinsicProcTable::Implementation::GetGenericIntrinsicName(
+    const std::string &name) const {
+  auto specificIntrinsic{specificFuncs_.find(name)};
+  if (specificIntrinsic != specificFuncs_.end()) {
+    if (const char *genericName{specificIntrinsic->second->generic}) {
+      return std::string(genericName);
```

Comment:
Is this ctor call needed? I think you could get away with `{genericName}` as an alternative but this may violate the coding style here.

Reply:
Changed to your suggestion, that also works and I think it is actually more in line with the coding style.

> If this PR is solving #257 can you please add the tests in the older [PR](https://github.com/Sameeranjoshi/llvm-project/commit/1944f799504eeed2df20c53b774f58405e5b1205), so we check `iabs` without passed as a function argument?
> 
> Thanks.

Sure, I have added the call tests also, thanks for making them.